File: 36. Segregate Even and Odd numbers/main.py
#User function Template for python3
class Solution:

    def segregateEvenOdd(self,arr):
        # code here
        i = 0
        j = len(arr) - 1

        while i <= j:
            if arr[i] & 1:
                arr[i], arr[j] = arr[j], arr[i]
                j -= 1
            else:
                i += 1

        arr[:i] = sorted(arr[:i])
        arr[i:] = sorted(arr[i:])

File: 36. Segregate Even and Odd numbers/main.cpp
class Solution {
  public:
    void segregateEvenOdd(vector<int>& arr) {
        // code here
        int i = 0;
        int j = arr.size() - 1;

        while (i <= j)
            if (arr[i] & 1)
            {
                int temp = arr[i];
                arr[i] = arr[j];
                arr[j] = temp;
                j--;
            }
            else
                i++;
        
        sort(arr.begin(), arr.begin() + i);
        sort(arr.begin() + i, arr.end());
    }
};